server/src/main/java/org/opensearch/index/remote/RemoteTranslogTransferTracker.java

```diff
@@ -0,0 +1,734 @@
+        downloadTimeMsMovingAverageReference = new AtomicReference<>(new MovingAverage(movingAverageWindowSize));
+    }
+
+    public long getTotalUploadsStarted() {
+        return totalUploadsStarted.get();
+    }
+
+    public long getTotalUploadsFailed() {
+        return totalUploadsFailed.get();
+    }
+
+    public long getTotalUploadsSucceeded() {
+        return totalUploadsSucceeded.get();
+    }
+
+    public long getUploadBytesStarted() {
+        return uploadBytesStarted.get();
+    }
+
+    public long getUploadBytesFailed() {
+        return uploadBytesFailed.get();
+    }
+
+    public long getUploadBytesSucceeded() {
+        return uploadBytesSucceeded.get();
+    }
+
+    public long getTotalUploadTimeInMillis() {
+        return totalUploadTimeInMillis.get();
+    }
+
+    ShardId getShardId() {
+        return shardId;
+    }
+
+    public void addUploadsStarted(long count) {
+        totalUploadsStarted.addAndGet(count);
+    }
+
+    public void addUploadsFailed(long count) {
+        checkTotal(totalUploadsStarted.get(), totalUploadsFailed.get(), totalUploadsSucceeded.get(), count);
+        totalUploadsFailed.addAndGet(count);
+    }
+
+    public void addUploadsSucceeded(long count) {
+        checkTotal(totalUploadsStarted.get(), totalUploadsFailed.get(), totalUploadsSucceeded.get(), count);
+        totalUploadsSucceeded.addAndGet(count);
+    }
+
+    public void addUploadBytesStarted(long count) {
+        uploadBytesStarted.addAndGet(count);
+    }
+
+    public void addUploadBytesFailed(long count) {
+        checkTotal(uploadBytesStarted.get(), uploadBytesFailed.get(), uploadBytesSucceeded.get(), count);
+        uploadBytesFailed.addAndGet(count);
+    }
+
+    public void addUploadBytesSucceeded(long count) {
+        checkTotal(uploadBytesStarted.get(), uploadBytesFailed.get(), uploadBytesSucceeded.get(), count);
+        uploadBytesSucceeded.addAndGet(count);
+    }
+
+    public void addUploadTimeInMillis(long duration) {
+        totalUploadTimeInMillis.addAndGet(duration);
+    }
+
+    public long getLastSuccessfulUploadTimestamp() {
+        return lastSuccessfulUploadTimestamp.get();
+    }
+
+    public void setLastSuccessfulUploadTimestamp(long lastSuccessfulUploadTimestamp) {
+        this.lastSuccessfulUploadTimestamp.set(lastSuccessfulUploadTimestamp);
+    }
+
+    boolean isUploadBytesMovingAverageReady() {
+        return uploadBytesMovingAverageReference.get().isReady();
+    }
+
+    public double getUploadBytesMovingAverage() {
+        return uploadBytesMovingAverageReference.get().getAverage();
+    }
+
+    public void updateUploadBytesMovingAverage(long count) {
+        synchronized (uploadBytesMutex) {
+            this.uploadBytesMovingAverageReference.get().record(count);
+        }
+    }
+
+    boolean isUploadBytesPerSecMovingAverageReady() {
+        return uploadBytesPerSecMovingAverageReference.get().isReady();
+    }
+
+    public double getUploadBytesPerSecMovingAverage() {
```

Comment:
this can be package private. Lets double check if we can make any other methods as package private if they are being accessed within the same package.